ResourceStatusSystem/Agent/EmailAgent.py

```diff
@@ -0,0 +1,149 @@
+''' EmailAgent
+
+  This agent sends an email every 30 minutes with aggregated actions taken for each site
+
+'''
+
+
+import os
+import json
+from datetime import datetime
+from DIRAC                                                       import S_OK, S_ERROR
+from DIRAC.Core.Base.AgentModule                                 import AgentModule
+from DIRAC.ResourceStatusSystem.Utilities                        import RssConfiguration
+from DIRAC.Interfaces.API.DiracAdmin                             import DiracAdmin
+
+__RCSID__ = '$Id: $'
+
+AGENT_NAME = 'ResourceStatus/EmailAgent'
+
+class EmailAgent( AgentModule ):
+
+  def __init__( self, *args, **kwargs ):
+
+    AgentModule.__init__( self, *args, **kwargs )
+    self.diracAdmin = DiracAdmin()
+
+    self.default_value = None
+    self.dirac_path = None
+    self.dirac_path = None
+    self.cacheFile = None
+
+  def initialize( self ):
+    ''' EmailAgent initialization
+    '''
+
+    self.default_value = '/home/smiras/DIRAC_Development/test_dir/'
+    #self.default_value = '/opt/dirac/pro/work/ResourceStatus/'
+    self.dirac_path = os.getenv('DIRAC', self.default_value)
+    self.cacheFile = self.dirac_path + 'cache.json'
+
+    return S_OK()
+
+  def execute( self ):
+
+    #if the file exists and it is not empty
+    if (os.path.isfile(self.cacheFile)) and (os.stat(self.cacheFile).st_size > 0):
+
+      time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+      #load the file
+      with open(self.cacheFile) as f:
```

Comment:
I don't know what is the general feeling about it, but I find it better when the opening mode is explicit